src/xbot/common/properties/DatabaseStorageBase.java

```diff
@@ -0,0 +1,173 @@
+            // The following method returns tables that match a given pattern. Since we don't care about
+            // Catalog, Schema, or Types (we only care about the TableNamePattern), we leave those
+            // fields null.
+            ResultSet tables = md.getTables(null, null, "PROPERTIES", null);
+
+            if (tables.next()) {
+                return true;
+            }
+            return false;
+
+        } catch (SQLException e) {
+            log.warn("Ran into problems when checking to see if the PROPERTIES table existed.");
+            log.warn(e.toString());
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    public boolean obliterateStorage() {
+        try {
+            Connection conn = DriverManager.getConnection(dbUrl);
+
+            if (propertiesTableExists(conn)) {
+                Statement sta = conn.createStatement();
+                String payload = "DROP TABLE PROPERTIES";
+
+                int response = sta.executeUpdate(payload);
+
+                if (response == 0) {
+                    return true;
+                }
+                // something went wrong
+                return false;
+            } else {
+                // table does not exist, nothing to obliterate
+                return true;
+            }
+        } catch (SQLException e) {
+            log.warn("Something went wrong when attempting to drop the PROPERTIES table!");
+            log.warn(e.toString());
+            e.printStackTrace();
+            return false;
+        }
+
+    }
+
+    protected String readFromFile() {
+        // Need to read out the database into a string
+
+        try {
+            Connection conn = DriverManager.getConnection(dbUrl);
+
+            if (propertiesTableExists(conn)) {
+                Statement sta = conn.createStatement();
+                String payload = "SELECT * FROM PROPERTIES";
+
+                ResultSet rs = sta.executeQuery(payload);
+
+                StringBuffer sb = new StringBuffer();
+
+                while (rs.next()) {
+                    sb.append(rs.getString("Type"));
+                    sb.append(propertyDelimiter);
+                    sb.append(rs.getString("Name"));
+                    sb.append(propertyDelimiter);
+                    sb.append(rs.getString("Value"));
+                    sb.append(lineSeperator);
+                }
+
+                return sb.toString();
+            } else {
+                // could not load properties table.
+                log.error("Could not find the properties table in the database when loading properties!");
+                log.error("All properties will be at their default levels!");
+            }
+        } catch (SQLException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+
+        return "";
+    }
+
+    protected void writeToFile(String data) {
+
+        Connection conn = null;
+
+        try {
```

Comment:
It seems like this connection try getting could be factored into it's own method? Also if it fails, do you really want to try and do the next thing?